src/core/server/server.ts

```diff
@@ -51,18 +58,21 @@ export class Server {
     // 2. When the process is run as dev cluster master in which case cluster manager
     // will fork a dedicated process where http service will be set up instead.
     let httpSetup: HttpServiceSetup | undefined;
-    const httpConfig = await this.http.config$.pipe(first()).toPromise();
+    const httpConfig = await this.configService
```

Comment:
I expected at least the linter to complain about access to private fields...ooook.

Reply:
Hmm, what linter should complain about? Public instance method accesses private property of the same instance or what you mean exactly?

Reply:
`config$` private member of `HttpService`  is accessed from outside

Reply:
`config$` was a property of `HttpModule` and I don't think it was private: `https://github.com/elastic/kibana/pull/33698/files#diff-56b9d98729b59445d4cdfe4f08e9dbacL37

Reply:
my bad, didn't see it's declared in constructor 👍